codalab/worker/dependency_manager.py

```diff
@@ -187,35 +261,48 @@ def loop(self):
         self._main_thread.start()
 
     def stop(self):
-        logger.info('Stopping local dependency manager')
+        logger.info('Stopping local dependency manager...')
         self._stop = True
         self._downloading.stop()
         self._main_thread.join()
-        logger.info('Stopped local dependency manager')
+        self._state_lock.release()
+        logger.info('Stopped local dependency manager.')
 
-    def _process_dependencies(self):
-        for dep_key, dep_state in self._dependencies.items():
-            with self._dependency_locks[dep_key]:
-                self._dependencies[dep_key] = self.transition(dep_state)
+    def _transition_dependencies(self):
+        with self._state_lock:
+            try:
+                dependencies, paths = self._fetch_state()
```

Comment:
Every call to fetch_state should not pass in a `default`. It should wrapped in:

```
try:
    dependencies, paths = self._fetch_state()
 except (ValueError, EnvironmentError):
                # Do nothing if an error is thrown while reading from the state file
                logging.traceback("failed to read state file")
                return
```

We can also remove the `default` argument in the implementation.